src/xtgeo/io/tsurf/_tsurf_reader.py

```diff
@@ -0,0 +1,799 @@
+        if axis_names not in cls.axis_names:
+            warnings.warn(
+                f"In file {filepath_errmsg}: Uncommon AXIS_NAME combination: "
+                f"{axis_names}. More common combinations are: "
+                f"{list(cls.axis_names)}",
+                UserWarning,
+                stacklevel=3,
+            )
+
+    @classmethod
+    def _validate_axis_units(
+        cls, axis_units: tuple[str, str, str], filepath_errmsg: str
+    ) -> None:
+        """Validate axis units tuple against known combinations.
+
+        Args:
+            axis_units: Tuple of axis units to validate
+            filepath_errmsg: Error context for meaningful messages
+        """
+
+        if len(axis_units) != 3 or any(unit == "" for unit in axis_units):
+            raise ValueError(
+                f"In file {filepath_errmsg}: AXIS_UNIT must have exactly three values, "
+                f"found {len(axis_units)}: {axis_units}"
+            )
+
+        # Check if the axis units are in our known common values
+        if axis_units not in cls.axis_units:
+            warnings.warn(
+                f"In file {filepath_errmsg}: Uncommon AXIS_UNIT combination: "
+                f"{axis_units}. More common combinations are: "
+                f"{list(cls.axis_units)}",
+                UserWarning,
+                stacklevel=3,
+            )
+
+    @classmethod
+    def _validate_zpositive(cls, zpositive: str, filepath_errmsg: str) -> None:
+        """Validate zpositive value and raise error for invalid values.
+
+        Args:
+            zpositive: Z-positive value to validate
+            filepath_errmsg: Error context for meaningful messages
+
+        Raises:
+            ValueError: If zpositive value is not in allowed values
+        """
+
+        if zpositive.lower() not in (z.lower() for z in cls.z_positives):
+            raise ValueError(
+                f"In file {filepath_errmsg}: Invalid ZPOSITIVE value '{zpositive}'. "
+                f"Allowed values are: {list(cls.z_positives)}"
+            )
+
+
+@dataclass
+class TSurfHeader:
+    name: str
+
+
+@dataclass
+class TSurfCoordSys:
+    name: str
+    axis_name: list[str]
+    axis_unit: list[str]
+    zpositive: str
+
+
+@dataclass
+class TSurfData:
+    header: TSurfHeader
+    coord_sys: Optional[TSurfCoordSys]
+    vertices: np.ndarray
+    triangles: np.ndarray
+
+
+def _read_line(
+    stream: Union[TextIOWrapper, StringIO],
+) -> Generator[list[str], Any, None]:
+    """
+    Iterate over lines from a TextIOWrapper or StringIO,
+    yielding lists of strings.
+
+    Filters out empty lines and comment lines.
+    """
+    for line in stream:
+        split_line = line.strip().split()
+
+        if not split_line or all(s == "" for s in split_line):
+            continue
+
+        # Skip comments
+        if split_line[0].startswith("#"):
+            continue
+
+        yield split_line
+
+
+def _is_tsurf_signature(line: list[str]) -> bool:
```

Comment:
See above; I think this can simply be replaced with a constant declaration

`TSURF_SIGNATURE = "GOCAD TSurf 1"`